app/models/signon_identity.rb

```diff
@@ -85,7 +88,32 @@ def to_session
       'full_name' => @full_name,
       'user_id' => @user.id,
       'profile_url' => @profile_url,
-      'logout_url' => @logout_url
+      'logout_url' => @logout_url,
+      'available_organisations' => @available_organisations,
+      'current_organisation' => @current_organisation
     }
   end
+
+  def current_estate
+    available_estates.find do |estate|
+      estate.sso_organisation_name == @current_organisation
+    end
+  end
```

Comment:
If we take this approach we need to protect against current_estate being nil – which I think it can become right now.